lib/timers.js

```diff
@@ -626,9 +636,6 @@ exports.clearImmediate = function(immediate) {
 
   immediate._onImmediate = undefined;
 
-  L.remove(immediate);
-
-  if (L.isEmpty(immediateQueue)) {
-    process._needImmediateCallback = false;
-  }
+  // leave queued, much faster overall to skip it later in processImmediate
```

Comment:
the result is very reproducible, it's 3x faster to set/clear without the remove.
It could be caching effects, ie during processing next item has already been prefetched
when the previous one was removed for processing.